wifistats/index.js

```diff
@@ -0,0 +1,99 @@
+/* global process */
+const exec          = require('child_process').exec;
+
+// The tools
+const airport       = require('./airport');
+const procwireless  = require('./procwireless');
+const netsh         = require('./netsh');
+
+let toolInstance;
+let supportWifiStats = true;
+
+/**
+ * Uses all available tools to query for wifi stats, and the one we found to
+ * work we store in global variable {@link toolInstance}, otherwise we set
+ * supportWifiStats to false.
+ * We call all tools in parallel and when all finishes we check the results.
+ *
+ * @return {Promise}
+ */
+function initTools() {
+    const os = process.platform;
+    let tool;
+
+    // check for running OS and use appropriate tool
+    if (os === 'win32') {
+        tool = netsh;
+    } else if (os === 'darwin') {
+        tool = airport;
+    } else if (os === 'linux') {
+        tool = procwireless;
+    }
```

Comment:
Is this what replaced the Promise.all that used to exist?

Reply:
Yes.

Reply:
My bad, I didn't realize you were making separate commits. I could have answered that myself easily but thanks for humoring me.

Reply:
I didn't mean to :) sorry

Reply:
I woke up and the first thing that came to mind was how good of a change this was to move away from Promise.all.